src/lex/cpp.rs

```diff
@@ -1066,15 +1066,33 @@ impl<'a> PreProcessor<'a> {
         local: bool,
         start: u32,
     ) -> Result<(), Locatable<Error>> {
-        let resolved = self.find_include_path(&filename, local, start)?;
-        let src = std::fs::read_to_string(&resolved)
-            .map_err(|err| Locatable {
-                data: CppError::IO(err.to_string()),
-                location: self.span(start),
-            })?
-            .into();
+        let (path, src) = match self.find_include_path(&filename, local, start) {
+            Ok(path) => {
+                let src = std::fs::read_to_string(&path)
+                    .map_err(|err| Locatable {
+                        data: CppError::IO(err.to_string()),
+                        location: self.span(start),
+                    })?
+                    .into();
+                (path, src)
+            }
+            Err(not_found) => {
+                let filename = match filename.file_name().and_then(|f| f.to_str()) {
+                    None => return Err(not_found),
+                    Some(f) => f,
+                };
+                match get_builtin_header(filename) {
+                    Some(file) => {
```

Comment:
Can you really override stddef.h? If so that's disgusting.

Reply:
My reasoning here is that if the standard library provides stddef.h, that's probably more accurate than the one I hacked together. The standard doesn't treat the compiler and library separately so it's up to us which we want to do.